subsys/net/lib/wifi_credentials/wifi_credentials_shell.c

```diff
@@ -290,3 +287,23 @@ SHELL_SUBCMD_ADD((wifi_cred), delete, NULL, "Delete network from storage.", cmd_
 SHELL_SUBCMD_ADD((wifi_cred), list, NULL, "List stored networks.", cmd_list_networks, 0, 0);
 
 SHELL_CMD_REGISTER(wifi_cred, &sub_wifi_cred, "Wi-Fi Credentials commands", NULL);
+
+#if CONFIG_WIFI_CREDENTIALS_CONNECT_STORED
+
+static int cmd_auto_connect(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_first_by_type(&NET_L2_GET_NAME(ETHERNET));
+	int rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, iface, NULL, 0);
+
+	if (rc) {
+		shell_error(sh,
+			    "An error occurred when trying to auto-connect to a network. err: %d",
+			    rc);
+	}
+	return 0;
+}
+
+SHELL_SUBCMD_ADD((wifi_cred), auto_connect, NULL, "Connect to any stored network.",
```

Comment:
If we change `(wifi_cred)` to `(wifi)` then the commands should be found under the wifi shell instead of creating a new shell.